tests/Doctrine/Tests/ORM/Functional/Ticket/DDC6393Test.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+class DDC6393Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(
+            [
+                $this->_em->getClassMetadata(A::class),
+                $this->_em->getClassMetadata(B::class)
+            ]
+        );
+    }
+
+    /**
+     * Test the the version of an entity can be fetched, when the id field and
+     * the id column are different.
+     */
+    public function testFetchVersionValueForDifferentIdFieldAndColumn()
+    {
+        $a = new A(1);
+        $this->_em->persist($a);
+
+        $b = new B($a, 'foo');
+        $this->_em->persist($b);        
+        $this->_em->flush();
+        
+        $this->assertSame(1, $b->getVersion());
+
+        $b->setSomething('bar');
+        $this->_em->flush();
+        
+        $this->assertSame(2, $b->getVersion());
+    }
+
+}
+
+/**
+ * @Entity
+ */
+class A
+{
+
+    /**
+     * @Id
+     * @Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @Version
+     * @Column(type="integer")
+     */
+    private $version;
+
+    public function __construct($id)
+    {
+        $this->id = $id;
+    }
+
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    public function getVersion()
```

Comment:
Please this method and make the property `public`, so that we simplify the entity a bit